diff -up go/src/cmd/go/internal/cfg/cfg.go.3~ go/src/cmd/go/internal/cfg/cfg.go
--- go/src/cmd/go/internal/cfg/cfg.go.3~	2021-07-14 13:48:38.080816805 +0200
+++ go/src/cmd/go/internal/cfg/cfg.go	2021-07-14 13:49:25.235864347 +0200
@@ -261,8 +261,8 @@ var (
 	GOPPC64  = envOr("GOPPC64", fmt.Sprintf("%s%d", "power", objabi.GOPPC64))
 	GOWASM   = envOr("GOWASM", fmt.Sprint(objabi.GOWASM))
 
-	GOPROXY    = envOr("GOPROXY", "https://proxy.golang.org,direct")
-	GOSUMDB    = envOr("GOSUMDB", "sum.golang.org")
+	GOPROXY    = envOr("GOPROXY", "direct,https://proxy.golang.org")
+	GOSUMDB    = envOr("GOSUMDB", "off")
 	GOPRIVATE  = Getenv("GOPRIVATE")
 	GONOPROXY  = envOr("GONOPROXY", GOPRIVATE)
 	GONOSUMDB  = envOr("GONOSUMDB", GOPRIVATE)
diff -up go/src/cmd/go/testdata/script/mod_sumdb_golang.txt.3~ go/src/cmd/go/testdata/script/mod_sumdb_golang.txt
--- go/src/cmd/go/testdata/script/mod_sumdb_golang.txt.3~	2021-07-14 13:48:38.081816828 +0200
+++ go/src/cmd/go/testdata/script/mod_sumdb_golang.txt	2021-07-14 13:50:33.723342865 +0200
@@ -2,12 +2,12 @@
 env GOPROXY=
 env GOSUMDB=
 go env GOPROXY
-stdout '^https://proxy.golang.org,direct$'
+stdout '^direct,https://proxy.golang.org$'
 go env GOSUMDB
-stdout '^sum.golang.org$'
+stdout '^off$'
 env GOPROXY=https://proxy.golang.org
 go env GOSUMDB
-stdout '^sum.golang.org$'
+stdout '^off$'
 
 # Download direct from github.
 [!net] skip
