diff --git a/src/cmd/godoc/godoc.go b/src/cmd/godoc/godoc.go
index f6dc678..3579dc7 100644
--- a/src/cmd/godoc/godoc.go
+++ b/src/cmd/godoc/godoc.go
@@ -81,8 +81,16 @@ var (
 	cmdHandler docServer
 	pkgHandler docServer
 )
+const (
+	SUSE_DOC_DIR = "/usr/share/doc/packages/go-doc/"
+	SUSE_SRC_DIR = "/usr/share/go/"
+)
 
 func initHandlers() {
+	// OPENSUSE: Bind godoc & source location
+	fs.Bind("/doc", OS(SUSE_DOC_DIR), "/doc", bindReplace)
+	fs.Bind("/src", OS(SUSE_SRC_DIR), "/src", bindBefore)
+
 	// Add named directories in -path argument as
 	// subdirectories of src/pkg.
 	for _, p := range filepath.SplitList(*pkgPath) {
